frontend/packages/ceph-storage-plugin/src/components/ocs-install/existing-cluster-modal.tsx

```diff
@@ -0,0 +1,69 @@
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { Trans, useTranslation } from 'react-i18next';
+import { match as RouteMatch } from 'react-router';
+import { Button } from '@patternfly/react-core';
+import { Modal } from '@console/shared/src/components/modal';
+import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager/src';
+import { history, resourcePathFromModel } from '@console/internal/components/utils';
+import { ListKind, referenceForModel } from '@console/internal/module/k8s';
+import { StorageClusterKind } from '../../types';
+import { OCSServiceModel } from '../../models';
+
+export const ExistingClusterModal: React.FC<ExistingClusterModalProps> = ({
+  match,
+  storageCluster,
+}) => {
+  const {
+    params: { ns, appName },
+  } = match;
+  const { t } = useTranslation();
+  const [isOpen, setIsOpen] = React.useState(true);
+
+  const clusterName = storageCluster.items.find((sc) => sc.status.phase !== 'Ignored').metadata
+    .name;
+
+  const storageClusterPath = `/k8s/ns/${ns}/clusterserviceversions/${appName}/${referenceForModel(
+    OCSServiceModel,
+  )}/${clusterName}`;
+
+  const onConfirm = () => {
+    setIsOpen(false);
+    history.push(resourcePathFromModel(ClusterServiceVersionModel, appName, ns));
+  };
+
+  const onCancel = () => {
+    setIsOpen(false);
+    history.push(storageClusterPath);
+  };
+
+  return (
+    <Modal
+      title={t('ceph-storage-plugin~Storage Cluster exists')}
+      titleIconVariant="warning"
+      isOpen={isOpen}
+      variant="small"
+      onClose={() => history.goBack()}
```

Comment:
```suggestion
      onClose={() => resourcePathFromModel(ClusterServiceVersionModel, appName, ns)}
```
Let's return them back to the OLM page no matter how they reach here.

Reply:
What do you mean by OLM page ?
Its returning to Installed Operator's page.

Reply:
what if I come here through a URL link? wouldn't we go back to the last url?

Reply:
yes, on closing the icon it will go and that's what we need to do. Its on closing the modal.
If you want to go to operator's page then you can click on "Go to operator's page"
Back to OLM page is explicit here. Refer the screenshot

Reply:
`history.goBack()` will refer to the last url, if I am not mistaken, so if I reach here from say the dashboard pages by changing the url. Wouldn't closing it lead me back to dashboard?

Reply:
@bipuladh here we are setting url to go back to the last page.
And here we are setting to operator and cluster page. see https://github.com/openshift/console/pull/7829/files#r569983381
A total of 3 urls.

Reply:
Exactly my concern. If the user closes it, do we really want to send them to the last URL or to the OLM details page for OCS Operator?

Reply:
Yes we want to send them to last url.
If he wants to go to operator page, he will choose operator page